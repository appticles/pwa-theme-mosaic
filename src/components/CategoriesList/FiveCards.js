import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import CategoryCard from '../CategoryCard/index';

const FiveCards = props => (
  <Grid columns={2} divided>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={Math.random()} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} tall={1} key={Math.random()} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} tall={1} key={Math.random()} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[3]} tall={1} key={Math.random()} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[4]} tall={1} key={Math.random()} />
      </GridColumn>
    </GridRow>
  </Grid>
);

FiveCards.propTypes = {
  categoriesList: PropTypes.array.isRequired,
};

export default FiveCards;
