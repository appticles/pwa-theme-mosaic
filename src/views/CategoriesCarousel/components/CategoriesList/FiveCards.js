import React from 'react';
import PropTypes from 'prop-types';
import { Grid, GridRow, GridColumn } from 'semantic-ui-react';

import { categoryPropType } from '../../reducer';
import CategoryCard from '../CategoryCard/index';

const FiveCards = props => (
  <Grid columns={2} divided className="one-third-row">
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[0]} key={props.categoriesList[0].id} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[1]} key={props.categoriesList[1].id} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[2]} key={props.categoriesList[2].id} />
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <CategoryCard category={props.categoriesList[3]} key={props.categoriesList[3].id} />
      </GridColumn>
      <GridColumn>
        <CategoryCard category={props.categoriesList[4]} key={props.categoriesList[4].id} />
      </GridColumn>
    </GridRow>
  </Grid>
);

FiveCards.propTypes = {
  categoriesList: PropTypes.arrayOf(categoryPropType).isRequired,
};

export default FiveCards;
