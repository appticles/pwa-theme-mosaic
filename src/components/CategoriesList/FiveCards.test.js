import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import FiveCards from './FiveCards';

describe('Five cards', () => {
  it('should render correctly', () => {
    const mockCategory = [
      {
        slug: 'mock-slug',
        id: 1,
        name: 'Some mock name',
      },
      {
        slug: 'mock-slug',
        id: 2,
        name: 'Some other mock name',
      },
      {
        slug: 'mock-slug',
        id: 3,
        name: 'Some name',
      },
      {
        slug: 'mock-slug',
        id: 4,
        name: 'Mock name',
      },
      {
        slug: 'mock-slug',
        id: 5,
        name: 'Some other other mock name',
      },
    ];
    const output = shallow(<FiveCards categoriesList={mockCategory} />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
