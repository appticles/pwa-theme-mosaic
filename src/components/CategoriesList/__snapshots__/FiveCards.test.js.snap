// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Five cards should render correctly 1`] = `
<Grid
  className="one-third-row"
  columns={2}
  divided={true}
>
  <GridRow>
    <GridColumn>
      <CategoryCard
        category={
          Object {
            "id": 1,
            "name": "Some mock name",
            "slug": "mock-slug",
          }
        }
        key="0.10891770873706386"
      />
    </GridColumn>
  </GridRow>
  <GridRow>
    <GridColumn>
      <CategoryCard
        category={
          Object {
            "id": 2,
            "name": "Some other mock name",
            "slug": "mock-slug",
          }
        }
        key="0.6567999356541583"
      />
    </GridColumn>
    <GridColumn>
      <CategoryCard
        category={
          Object {
            "id": 3,
            "name": "Some name",
            "slug": "mock-slug",
          }
        }
        key="0.9934359151482539"
      />
    </GridColumn>
  </GridRow>
  <GridRow>
    <GridColumn>
      <CategoryCard
        category={
          Object {
            "id": 4,
            "name": "Mock name",
            "slug": "mock-slug",
          }
        }
        key="0.3908024946023123"
      />
    </GridColumn>
    <GridColumn>
      <CategoryCard
        category={
          Object {
            "id": 5,
            "name": "Some other other mock name",
            "slug": "mock-slug",
          }
        }
        key="0.9610344680213294"
      />
    </GridColumn>
  </GridRow>
</Grid>
`;
